### PR TITLE
fix(dockerfile): analyze all FROM lines via batch analysis

### PR DESCRIPTION
**Pull request by a-oren**
## Summary
- Change DockerfileProvider from analyzing only the last FROM line to analyzing **all** FROM lines in multi-stage Dockerfiles
- Add `batch` flag to `Provider.Content` to route between `/api/v5/analysis` and `/api/v5/batch-analysis` endpoints
- Resolve ARG substitutions (both `${VAR}` and `$VAR` syntax) using default values; skip `FROM scratch` and unresolvable ARGs
- Route batch content through `ExhortApi` for component, stack, and license analysis flows
- Fix case-insensitive `scratch` comparison so `FROM SCRATCH`, `FROM Scratch`, etc. are properly skipped ([TC-5098](https://redhat.atlassian.net/browse/TC-5098))
- Fix ARG pattern to capture quoted and space-containing default values ([TC-5167](https://redhat.atlassian.net/browse/TC-5167))
- Fix Docker Hub library-prefix normalization to use `repositoryUrl.substring()` instead of `lower.substring()` for consistent casing ([TC-5168](https://redhat.atlassian.net/browse/TC-5168))

## Test plan
- [x] 26 unit tests pass (parsing, batch content, ecosystem integration, provider properties, case-insensitive scratch)
- [x] 3 new Dockerfile ARG quoting tests (double-quoted, single-quoted, space-containing values)
- [x] 2 new ImageRef PURL normalization tests (library-prefix/bare-name parity, consistent casing)
- [x] Spotless formatting check passes
- [x] Full test suite passes

Fixes: [TC-5071](https://redhat.atlassian.net/browse/TC-5071)
Fixes: [TC-5098](https://redhat.atlassian.net/browse/TC-5098)
Fixes: [TC-5168](https://redhat.atlassian.net/browse/TC-5168)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[TC-5098]: https://redhat.atlassian.net/browse/TC-5098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5167]: https://redhat.atlassian.net/browse/TC-5167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5168]: https://redhat.atlassian.net/browse/TC-5168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5071]: https://redhat.atlassian.net/browse/TC-5071?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/io/github/guacsec/trustifyda/providers/DockerfileProvider.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          if (image.contains("$")) {
+            image = resolveArgSubstitutions(image, argDefaults);
+            if (image == null) {
+              LOG.info(
+                  String.format(
+                      "Skipping FROM line with unresolvable ARG in %s: %s", dockerfile, tokens[i]));
+              continue;
+            }
+          }
+          if ("scratch".equals(image)) {
```
[the remark this answers is not in the document]
Comment (a-oren) [reply src/main/java/io/github/guacsec/trustifyda/providers/DockerfileProvider.java]: [sdlc-workflow/verify-pr] Classified as **code change request** — sub-task [TC-5098](https://redhat.atlassian.net/browse/TC-5098) created to address this feedback.

File: src/main/java/io/github/guacsec/trustifyda/image/ImageRef.java [hunk 2]
```diff
@@ -140,11 +140,26 @@ void checkImageDigest() {
...
   public PackageURL getPackageURL() throws MalformedPackageURLException {
     TreeMap<String, String> qualifiers = new TreeMap<>();
     var repositoryUrl = this.image.getNameWithoutTag();
     var simpleName = this.image.getSimpleName();
+
+    // Normalize Docker Hub image references so all forms produce the same PURL:
+    //   node                    → docker.io/node
+    //   docker.io/library/node  → docker.io/node
+    if (repositoryUrl != null) {
+      var lower = repositoryUrl.toLowerCase();
```
[the remark this answers is not in the document]
Comment (a-oren) [reply src/main/java/io/github/guacsec/trustifyda/image/ImageRef.java:155]: [sdlc-workflow/verify-pr] Classified as **suggestion** — this proposes using Locale.ROOT for toLowerCase(), which is a general best practice but has no established codebase pattern (20+ existing uses of toLowerCase() without Locale.ROOT, zero uses of Locale.ROOT). Not documented in CONVENTIONS.md. No sub-task created.

File: src/test/java/io/github/guacsec/trustifyda/providers/Dockerfile_Provider_Test.java [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+        // Then
+        assertThat(content.batch).isTrue();
+        assertThat(content.type).isEqualTo(Api.CYCLONEDX_MEDIA_TYPE);
+
+        JsonNode batchJson = MAPPER.readTree(content.buffer);
+        assertThat(batchJson.isObject()).isTrue();
+        assertThat(batchJson.size()).isEqualTo(2);
+        assertThat(batchJson.has(nodePurl.toString())).isTrue();
+        assertThat(batchJson.has(nginxPurl.toString())).isTrue();
```
[the remark this answers is not in the document]
Comment (a-oren) [reply src/test/java/io/github/guacsec/trustifyda/providers/Dockerfile_Provider_Test.java:402]: [sdlc-workflow/verify-pr] Classified as **suggestion** — this proposes additional test coverage for the all-images-fail SBOM generation path. While reasonable, CONVENTIONS.md does not require exhaustive error-path coverage, and the existing test suite already covers the error path in parseAllFromImages (all_invalid scenario). No sub-task created.

**Comment by a-oren** [comment]
[sdlc-workflow/verify-pr] Re: @sourcery-ai[bot] review — Classified as **suggestion** — (1) scratch case-sensitivity: addressed via sub-task [TC-5098](https://redhat.atlassian.net/browse/TC-5098) created from the inline comment. (2) buildAnalysisUri URI building: this proposes using a URI builder instead of string concatenation, which is a valid robustness suggestion but has no established codebase pattern or CONVENTIONS.md backing. No sub-task created for item (2).

[TC-5098]: https://redhat.atlassian.net/browse/TC-5098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by a-oren** [comment]
## Verification Report for TC-5071 (commit 3b43c23)

| Check | Result | Details |
|-------|--------|---------|
| Review Feedback | WARN | 1 code change request → sub-task [TC-5098](https://redhat.atlassian.net/browse/TC-5098) created (scratch case-insensitive comparison) |
| Root-Cause Investigation | DONE | Convention gap identified → [TC-5099](https://redhat.atlassian.net/browse/TC-5099) created (document case-sensitivity consistency convention) |
| Scope Containment | WARN | All 3 task-specified files present; 18 additional files (test fixtures, supporting classes, bundled TRUSTIFY_DA_RECOMMEND feature) |
| Diff Size | WARN | 935+/29- across 23 files; includes bundled second feature commit beyond task scope |
| Commit Traceability | WARN | 1/3 commits references TC-5071; others reference TC-4938 or no task ID |
| Sensitive Patterns | PASS | No secrets or credentials detected |
| CI Status | PASS | All 44 CI checks pass |
| Acceptance Criteria | PASS | All 6 acceptance criteria satisfied |
| Test Quality | WARN | Repetitive Test Detection: 2 recommend tests share ~90% boilerplate; Test Documentation: inconsistent JavaDoc (Dockerfile_Provider_Test has docs, ImageRefTest/Exhort_Api_Test do not); Eval Quality: N/A |
| Test Change Classification | ADDITIVE | 30 new test methods added, 0 removed |
| Verification Commands | PASS | `./mvnw test` passes (488 tests, 6 pre-existing Python env errors on main); `./mvnw spotless:check` passes |

### Overall: WARN

**Issues requiring attention:**
- **Code fix needed:** `"scratch".equals(image)` should be `"scratch".equalsIgnoreCase(image)` in `DockerfileProvider.java:140` — sub-task [TC-5098](https://redhat.atlassian.net/browse/TC-5098) tracks this fix
- **Scope note:** PR bundles commit `63309e25` (TRUSTIFY_DA_RECOMMEND env var feature) which is unrelated to TC-5071. Consider splitting into a separate PR for clean traceability
- **Commit traceability:** Only 1/3 commits references TC-5071 — commit `bf0257f5` references TC-4938, commit `63309e25` has no task reference

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.12.2.*

[TC-5098]: https://redhat.atlassian.net/browse/TC-5098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5099]: https://redhat.atlassian.net/browse/TC-5099?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5098]: https://redhat.atlassian.net/browse/TC-5098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Review by ruromero — CHANGES_REQUESTED** [review]
## Review Summary

The PR adds Dockerfile/Containerfile support by parsing FROM lines, generating per-image SBOMs via Syft, and sending them to the batch-analysis endpoint. The Dockerfile parsing logic (ARG resolution, scratch skipping, multi-stage support) is solid and well-tested. However, the analysis routing has an architectural issue that breaks parity with how both IDE extensions currently consume image results.

---

### Critical: Dockerfile analysis must route through `imageAnalysis()`, not `stackAnalysis()`

Both IDE extensions currently handle Dockerfiles by calling `imageAnalysis(Set<ImageRef>)`, which returns `Map<ImageRef, AnalysisReport>` — one report per FROM image:

- **IntelliJ** (`image/ApiService.java:69`): `exhortApi.imageAnalysis(imageRefs)`
- **VS Code** (`rhda.ts:116-117`): `fileType === 'docker'` → `executeDockerImageAnalysis()` → `exhort.imageAnalysis()`

This PR routes Dockerfile content through `stackAnalysis()`/`componentAnalysis()` instead, which return `CompletableFuture<AnalysisReport>` (single report). The batch results are collapsed with:

```java
reports.values().iterator().next(); // takes only the FIRST image's report
```

For a multi-FROM Dockerfile, all image reports after the first are silently dropped.

**Suggestion:** Keep `stack Dockerfile` as a valid CLI command, but internally detect that the manifest is a Dockerfile and delegate to the existing `imageAnalysis(Set<ImageRef>)` path. The `DockerfileProvider` parsing logic (FROM extraction, ARG resolution) is valuable — it should produce `Set<ImageRef>` that feeds into `imageAnalysis()`, preserving the `Map<ImageRef, AnalysisReport>` return type.

This also eliminates the duplicated batch handling code in `stackAnalysis` (inline) and `getBatchAnalysisReportForComponent` — the existing `imageAnalysis` already does all of this correctly.

Additionally, `stackAnalysisHtml()` was not updated for batch content at all (still uses `buildStackRequest`), so generating an HTML report for a Dockerfile via this path would not use the batch endpoint.

---

### Bug: Docker Hub normalization casing inconsistency in `ImageRef.getPackageURL()`

The library-prefix branch uses `lower.substring()` which lowercases the image name, while the bare-name branch preserves original casing via `simpleName`:

```java
if (lower.equals(simpleName.toLowerCase())) {
    repositoryUrl = "docker.io/" + simpleName;           // preserves casing
} else if (lower.startsWith(DOCKER_HUB_LIBRARY_PREFIX)) {
    repositoryUrl = "docker.io/" + lower.substring(...);  // lowercased
}
```

`docker.io/library/MyImage` → `docker.io/myimage` (lowercased)
`MyImage` → `docker.io/MyImage` (preserved)

Both should behave the same. The library-prefix branch should use `repositoryUrl.substring(DOCKER_HUB_LIBRARY_PREFIX.length())` to preserve original casing, consistent with the bare-name branch. (Note: `repositoryUrl` is later lowercased when put into qualifiers anyway, but the PURL `name` field should be consistent.)

---

### Low: Recommend env var naming mismatch across clients

Java uses `TRUSTIFY_DA_RECOMMEND`, JavaScript uses `TRUSTIFY_DA_RECOMMENDATIONS_ENABLED`. These should be aligned so users don't need different env vars per client. The JS client's name is already established, so consider adopting `TRUSTIFY_DA_RECOMMENDATIONS_ENABLED` in the Java client as well.

Also note the semantic inversion: Java defaults to `true` (recommend enabled unless explicitly set to false), while JS checks for the string `'false'`. They happen to have the same default behavior, but the var names should match.

---

### Low: Silent license check skip for batch content

```java
if (!isLicenseCheckEnabled() || content.batch) {
    return CompletableFuture.completedFuture(new ComponentAnalysisResult(report, null));
}
```

License checking is silently skipped when `content.batch` is true, with no logging. This is likely intentional (license check doesn't apply to images), but a debug log would help troubleshooting.

---

### Positive

- Dockerfile parsing is well-structured: correct handling of multi-stage builds, ARG substitution, scratch skipping, and case-insensitive FROM matching
- Excellent test coverage: 26 tests for DockerfileProvider with 13 fixture Dockerfiles covering edge cases
- The `Ecosystem.isDockerfile()` correctly requires a dot separator, avoiding false matches on files like `Dockerfileignore`

**Comment by a-oren** [comment]
[sdlc-workflow/verify-pr] Re: @ruromero review —

1. **Critical: Dockerfile analysis must route through imageAnalysis()** — Classified as **code change request**. Addressed by commit bdc3e41 which routes Dockerfile manifests through `executeImageAnalysis(Set<ImageRef>)` in `App.java`, preserving per-image `Map<ImageRef, AnalysisReport>` output. `DockerfileProvider.parseImageRefs()` converts FROM lines to `Set<ImageRef>` for the existing `imageAnalysis()` path. No sub-task needed — fix already committed.

2. **Bug: Docker Hub normalization casing inconsistency** — Classified as **code change request**. Sub-task [TC-5168](https://redhat.atlassian.net/browse/TC-5168) created to fix `lower.substring()` → `repositoryUrl.substring()` in `ImageRef.getPackageURL()` line 159.

3. **Low: Recommend env var naming mismatch** — Classified as **suggestion** — cross-client env var alignment (`TRUSTIFY_DA_RECOMMEND` vs `TRUSTIFY_DA_RECOMMENDATIONS_ENABLED`) is a valid concern but represents a broader cross-repository decision, not a convention violation in this PR. No sub-task created.

4. **Low: Silent license check skip for batch content** — Classified as **suggestion** — the existing `!isLicenseCheckEnabled()` check was already silent before this PR; adding debug logging is optional. No sub-task created.

[TC-5168]: https://redhat.atlassian.net/browse/TC-5168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by a-oren** [comment]
## Verification Report for TC-5071 (commit bdc3e41)

| Check | Result | Details |
|-------|--------|---------|
| Review Feedback | WARN | 2 code change requests from ruromero: (1) routing through imageAnalysis — addressed by commit bdc3e41; (2) Docker Hub normalization bug — sub-task [TC-5168](https://redhat.atlassian.net/browse/TC-5168) created |
| Root-Cause Investigation | DONE | implement-task skill gap identified → [TC-5169](https://redhat.atlassian.net/browse/TC-5169) created (normalization-consistency check for parallel code branches) |
| Scope Containment | WARN | All 3 task-specified files present; 23 additional files (test fixtures, supporting classes, bundled features) |
| Diff Size | PASS | 26 files changed is proportionate for new provider feature with batch analysis and test fixtures |
| Commit Traceability | WARN | 2/5 commits reference TC-5071; others reference TC-4938, TC-5098, or no task ID |
| Sensitive Patterns | PASS | No secrets or credentials detected |
| CI Status | PASS | All 45 CI checks pass |
| Acceptance Criteria | PASS | 6 of 6 criteria satisfied |
| Test Quality | WARN | Repetitive Test Detection: 2 test pairs in AppTest.java are parameterization candidates; Test Documentation: PASS (Dockerfile_Provider_Test has JavaDoc, other files match existing no-JavaDoc convention); Eval Quality: N/A |
| Test Change Classification | ADDITIVE | 30+ new test methods added across 4 files, 0 removed |
| Verification Commands | PASS | `./mvnw test` passes (577 non-Python tests); `./mvnw spotless:check` passes |

### Overall: WARN

**Issues requiring attention:**
- **Code fix needed:** Docker Hub normalization casing inconsistency in `ImageRef.getPackageURL()` line 159 — `lower.substring()` should be `repositoryUrl.substring()` to preserve casing. Sub-task [TC-5168](https://redhat.atlassian.net/browse/TC-5168) tracks this fix.
- **Routing fix applied:** Dockerfile analysis now routes through `imageAnalysis()` (commit bdc3e41), preserving per-image `Map<ImageRef, AnalysisReport>` output as requested by ruromero.
- **Scope note:** PR bundles commits from TC-4938 and the TRUSTIFY_DA_RECOMMEND feature (commit 63309e25) beyond TC-5071 scope.
- **Prior sub-task:** [TC-5098](https://redhat.atlassian.net/browse/TC-5098) (scratch case-insensitive comparison) — addressed by commit 76a3a72.

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.13.0.*

[TC-5168]: https://redhat.atlassian.net/browse/TC-5168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5169]: https://redhat.atlassian.net/browse/TC-5169?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TC-5168]: https://redhat.atlassian.net/browse/TC-5168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java [hunk 4]
```diff
@@ -101,6 +101,7 @@ public final class ExhortApi implements Api {
   public static final String S_API_V5_LICENSES = "%s/api/v5/licenses/%s";
   public static final String S_API_V5_LICENSES_IDENTIFY = "%s/api/v5/licenses/identify";
   private static final String TRUSTIFY_DA_LICENSE_CHECK = "TRUSTIFY_DA_LICENSE_CHECK";
+  private static final String TRUSTIFY_DA_RECOMMEND = "TRUSTIFY_DA_RECOMMEND";
```
Comment (Strum355) [inline src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java:104]: The JS client uses `TRUSTIFY_DA_RECOMMENDATIONS_ENABLED` afaik for the same feature, so this should match that

File: src/main/java/io/github/guacsec/trustifyda/providers/DockerfileProvider.java [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+ */
+public final class DockerfileProvider extends Provider {
+
+  private static final Logger LOG = LoggersFactory.getLogger(DockerfileProvider.class.getName());
+
+  private static final Pattern FROM_LINE_PATTERN =
+      Pattern.compile("^FROM\\s+", Pattern.CASE_INSENSITIVE);
+
+  private static final Pattern ARG_LINE_PATTERN =
+      Pattern.compile("^ARG\\s+([A-Za-z_][A-Za-z0-9_]*)=(\\S+)", Pattern.CASE_INSENSITIVE);
```
Comment (Strum355) [inline src/main/java/io/github/guacsec/trustifyda/providers/DockerfileProvider.java]: `\\S+` truncates quoted ARG defaults at the first space. `ARG MSG="hello world"` captures only "hello as the default value, and the quotes are included in the captured text for any quoted value.

Suggested fix: capture everything after = to end of line, then strip surrounding quotes: `Pattern.compile("^ARG\\s+([A-Za-z_][A-Za-z0-9_]*)=(.+)", Pattern.CASE_INSENSITIVE);` with a post-capture strip in the consumer:
```
 String val = argMatcher.group(2);
 if ((val.startsWith("\"") && val.endsWith("\"")) || (val.startsWith("'") && val.endsWith("'"))) {
     val = val.substring(1, val.length() - 1);
 }
 argDefaults.put(argMatcher.group(1), val);
```
Ideally we'd have some test cases to cover the varying forms of this (newlines, escape sequences, different quotes etc)

**Review by Strum355 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java (hunk 4 above)
Comment (a-oren) [reply src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java:104]: this is from an older worktree (TC-4694). The JS client's main branch has already been updated to use TRUSTIFY_DA_RECOMMEND

Comment (a-oren) [reply src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java:104]: [sdlc-workflow/verify-pr] Classified as: **Question**

Investigation: Both Java and JS clients use `TRUSTIFY_DA_RECOMMEND` — the name is consistent. The JS client does NOT use `TRUSTIFY_DA_RECOMMENDATIONS_ENABLED`.

File: src/main/java/io/github/guacsec/trustifyda/providers/DockerfileProvider.java (hunk 5 above)
Comment (a-oren) [reply src/main/java/io/github/guacsec/trustifyda/providers/DockerfileProvider.java]: [sdlc-workflow/verify-pr] Classified as: **Code change request**

Status: Fixed in commit 4ed519a — ARG_LINE_PATTERN changed from `\\S+` to `.+` and surrounding quotes are stripped.

**Comment by a-oren** [comment]
## Verification Report for TC-5071 (commit fe2c973)

| Check | Result | Details |
|-------|--------|---------|
| Review Feedback | PASS | 7 items classified (1 code change request — already fixed in 4ed519a, 1 question, 5 suggestions); no sub-tasks needed |
| Root-Cause Investigation | N/A | No sub-tasks created |
| Scope Containment | WARN | All task-specified files present; 4 additional files (ImageRef.java, ImageRefTest.java, Exhort_Api_Test.java, AppTest.java) support the feature |
| Diff Size | PASS | 1,285 additions / 34 deletions across 29 files — proportionate for new provider + comprehensive tests |
| Commit Traceability | WARN | 1/8 commits references TC-5071; others reference TC-4938, TC-5098, TC-5168, or have no task reference |
| Sensitive Patterns | PASS | No secrets, credentials, or sensitive data detected |
| CI Status | PASS | All CI checks pass (build, 42 integration tests, run-tests) |
| Acceptance Criteria | PASS | All 14 acceptance criteria satisfied |
| Test Quality | WARN | Repetitive Test Detection: WARN (ARG-resolution and single-FROM test groups could be parameterized); Test Documentation: PASS; Eval Quality: N/A |
| Test Change Classification | ADDITIVE | All test changes add new tests or expand existing setup; no tests removed or weakened |
| Verification Commands | N/A | No verification commands specified |

### Overall: WARN

**Items for reviewer attention:**

- **Scope Containment (WARN):** PR includes `ImageRef.java` changes (Docker Hub normalization, TC-5168) and `Exhort_Api_Test.java` additions (TRUSTIFY_DA_RECOMMEND tests) beyond the TC-5071 task spec. These are reasonable supporting changes but represent multi-task bundling.

- **Commit Traceability (WARN):** This PR bundles work from 4 Jira tasks (TC-4938, TC-5071, TC-5098, TC-5168). Only commit 3b43c23 directly references TC-5071. Commits 63309e2, bdc3e41, 4ed519a, and fe2c973 have no Jira references.

- **Repetitive Test Detection (WARN):** `Dockerfile_Provider_Test.java` contains two groups of structurally identical tests that could benefit from `@ParameterizedTest`:
  - 5 ARG-resolution tests (all assert single `"ubuntu:22.04"` result from different fixture files)
  - 6 single-FROM tests (all assert single-element list with different expected images)

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/RHEcosystemAppEng/sdlc-plugins) v0.13.2.*

**Review by ruromero — APPROVED** [review]

**Review by Strum355 — APPROVED** [review]
